Rm all legacy

lukpueh opened the pull request:
Fixes #1745 

- Removes legacy code, tests and docs
- Updates remaining docs and some project metadata accordingly


**Preliminary work**
  - https://github.com/python/peps/pull/2240
  - https://github.com/theupdateframework/theupdateframework.io/pull/31
  - https://github.com/theupdateframework/python-tuf/pull/1773
  - https://github.com/theupdateframework/python-tuf/pull/1769
  - https://github.com/theupdateframework/python-tuf/pull/1768
  - https://github.com/theupdateframework/python-tuf/pull/1767
 - Updated references in [CII Badge App](https://bestpractices.coreinfrastructure.org/de/projects/1351)
 - Checked specification for stale references to python-tuf (found none)


**Note to reviewers**:
- Excluding deleted files from diff should make review easier
- I'd appreciate a quick look through the remaining files in the repo, if I missed anything.
- Check commit messages for details


**Note of appreciation**
Many thanks to everyone who helped python-tuf to evolve! ❤️ 

Most notably (based on prior [Acknowledgements](https://github.com/theupdateframework/python-tuf/blame/b2704a56a8c5023ffd7d377d8a8e72fb26a46a8a/README.md#L130-L132) and *git shortlog* and in alphabetical order):

*Sebastien Awwad, Justin Cappos, Geremy Condra, Vladimir Diaz, Zane Fisher, Pankhuri Goyal, Jussi Kukkonen, Trishank Kuppusamy, Joshua Lock, Marina Moore, Justin Samuel, Teodora Sechkova, Tian Tian, Santiago Torres-Arias, Martin Vrachev, Yuyu Zheng, Kairo de Araujo*

lukpueh commented:
Just moved out test updates for unittest_toolbox removal in this PR to #1792 to make reviewing easier. That PR also fixes the failing windows run seen in this PR.

Commit:
doc: don't mention legacy tools in main README

Legacy tools will be removed in subsequent commits.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
doc: drop documentation for legacy tools

Remove documentation for legacy client, repository/developer tool
and command line tools, which will be removed in subsequent
commits.

See #1797 and #1798 for replacing ATTACKS.md and QUICKSTART.md.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
test: drop legacy test metadata generation scripts

Remove metadata generation scripts based on legacy
repository/developer tools, which will be removed in subsequent
commits.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
test: drop usage of custom log module in new test

Old tests are not touched as they will be removed in subsequent
commits, along with the custom log module.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
test: drop legacy tests and test tooling

Remove tests for legacy client, repository/developer tool and
command line tools, which will be removed in subsequent commits.

This commits also removes obsolete test tooling:
- Regarding simple_https_server + test certificates -- http/https
is no longer handled by tuf client directly but transparently by
the underlying requests module used by the default fetcher
implementation.
- For details about unittest_toolbox see #1792

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
Drop legacy implementation

Remove legacy client, repository/developer tool, command line
tools, and underlying libraries and utilities.

See docs/1.0.0-ANNOUNCEMENT.md for details about their replacement,
deprecation strategy and migration instructions.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

Commit:
Update misc repo metadata after legacy code drop

- Update linter config to no longer distinguish between legacy
and new implementation. This requires addressing a linter warning
in an until now not linted module (tuf/__init__.py).

- Remove obsolete rules in MANIFEST.in (source distribution) and
tests/.coveragerc (test coverage).

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

jku reviewed (commented):
Looks great :grin:  I might have to take another look though, it's hard to cover everything in one go.

Some observations:
* files in examples/repo_example/ reference repository_tool and repository_lib: they could be updated to indicate that those have now been deprecated
* ATTACKS.md would indeed be nice to have but that's clearly a project of its own and probably won't look much like the old one so I'm fine with leaving that for now

lukpueh commented:
> files in examples/repo_example/ reference repository_tool and repository_lib: they could be updated to indicate that those have now been deprecated

Thanks for catching! This was on my mind at some point.. but then it wasn't. :D  Will push another commit here.

jku reviewed (approved):
looks good to me. The only additional potential removal I found was the outdated AUTHORS.txt but that does not need to be in this PR.

EDIT: and the test data generation issue I just filed #1806

jku commented:
```
$ tox -e py
...
Ran 128 tests in 2.058s
```
this is so good.

(if anyone is wondering why there are so many fewer tests: we're using a lot more subtests and those are not counted individually: my Stetson-Harrison assessment is that the new code on average has better coverage than the old one did, although specific areas may be worse).

MVrachev commented on tox.ini:
```diff
@@ -38,17 +38,12 @@ commands =
 [testenv:lint]
 changedir = {toxinidir}
-lint_dirs = tuf/api tuf/ngclient examples tests
+lint_dirs = tuf examples tests
```
Did you check if all of the tools recursively check the next folders in `tuf/` - `tuf/ngclient` and `tuf/api`?

MVrachev reviewed (approved):
I was waiting for this pr for a long time!
Great stuff here.

I totally forgot that the old updater is 3070 lines, repository_lib 2306 lines, and repository_tool 3291... wow

Also, thanks for the last commitment! I was planning on doing that.

Only one question: do we have a test for key revocation?

jku commented:
> Only one question: do we have a test for key revocation?

In the sense that the old tests used the word "revocation", yes we do.

test_updater_key_rotations.py contains loads of different cases of key rotations: some remove ("revoke") old keys, some don't. test_updater_top_level_update.py has tests for fast forward recovery which include complete key rotations (these are made easier to read in #1766).

jku commented:
I should have stated this explicitly: 
* I think the PR is good and we should deal with any fallout with new issues after merging
* I did not merge myself as I wanted other maintainers to have time to review as it's a major change

Commit:
doc: update deprecation info in repo_example docs

Rephrase deprecation info in repo_example modules doc headers
to reflect that the deprecation has happened.

Signed-off-by: Lukas Puehringer <lukas.puehringer@nyu.edu>

lukpueh commented:
Thanks for your reviews, @jku and @MVrachev!


> * files in examples/repo_example/ reference repository_tool and repository_lib: they could be updated to indicate that those have now been deprecated

Fixed in 9816c40.

> Did you check if all of the tools recursively check the next folders in `tuf/` - `tuf/ngclient` and `tuf/api`?

Yes, I did. And yes, they do.

> Only one question: do we have a test for key revocation?

I admit that I didn't question test completeness. I just `rm tests/test_*_old.py` (relying on 22fe1e69e4315ce3e96d712fbf8d50871ce5f64f) and cross-checked with effective test coverage, which didn't decrease.

lukpueh commented:
> I should have stated this explicitly:
> 
> * I think the PR is good and we should deal with any fallout with new issues after merging
> * I did not merge myself as I wanted other maintainers to have time to review as it's a major change

@-'other (active) maintainers', i.e. @mnm678, @joshuagl, mind wrapping this up? 🚀 😃

mnm678 reviewed (approved):
Thanks to all the people who worked on the new client! This is really exciting, and looks good to me.

lukpueh commented:
Thanks for the reviews, everyone! Merging...
